### PR TITLE
Merge legacy and mining /blocks components and APIs

### PR DESCRIPTION
**Pull request by nymkappa**
Fixes https://github.com/mempool/mempool/issues/1692

This PR completely gets rid of the legacy `/blocks` page (currently seen when the mining dashboard is disabled in the frontend config, or for non Bitcoin networks) and its related `/blocks` node backend API.

There is now one single `blocks-list` component which properly display what is available according to the currently running environment.

The legacy `/blocks` API has been deleted, and the `/blocks-extras` API has been renamed as `/blocks`.

### CODE REVIEW

**Comment by softsimon** [comment]
The `mining-dashboard.component.html` and `dashboard.component.html` still do routerLink to `/mining/blocks`

My latest block widget is suddenly very long vertically

Screen Shot 2022-05-20 at 22 19 42

File: frontend/src/app/components/blocks-list/blocks-list.component.html [hunk 1]
```diff
@@ -1,29 +1,30 @@
...
       <thead>
-        <th class="height text-left" [class]="widget ? 'widget' : ''" i18n="latest-blocks.height">Height</th>
-        <th class="pool text-left" [class]="widget ? 'widget' : ''" i18n="mining.pool-name">Pool</th>
-        <th class="timestamp" i18n="latest-blocks.timestamp" *ngIf="!widget">Timestamp</th>
-        <th class="mined" i18n="latest-blocks.mined" *ngIf="!widget">Mined</th>
-        <th class="reward text-right" i18n="latest-blocks.reward" [class]="widget ? 'widget' : ''">Reward</th>
-        <th class="fees text-right" i18n="latest-blocks.fees" *ngIf="!widget">Fees</th>
-        <th class="txs text-right" i18n="dashboard.txs" [class]="widget ? 'widget' : ''">TXs</th>
-        <th class="size" i18n="latest-blocks.size" *ngIf="!widget">Size</th>
+        <th class="height text-left" [class]="widget ? 'widget' : ''" [class]="indexingAvailable ? '' : 'legacy'" i18n="latest-blocks.height">Height</th>
```
Comment (softsimon) [inline frontend/src/app/components/blocks-list/blocks-list.component.html:11]: Maybe it's not possible to combine multiple `[class]` without conflict.

Instead consider:

`[ngClass]="{'widget: widget, 'legacy': !indexingAvailable}"`

Comment (nymkappa) [reply frontend/src/app/components/blocks-list/blocks-list.component.html:11]: That fixed it, thanks!

**Comment by nymkappa** [comment]
* Added redirection from `/mining/blocks` to `/blocks` on bitcoin mainnet/tesnet/signet in routing module
* Fixed [class] conflict using @softsimon suggestion
* Update outdated [router] link so they directly point to `/blocks`

**Comment by nymkappa** [comment]
Rebased on master

**Comment by softsimon** [comment]
Seems to break on Liquid.

`liquid/api/v1/blocks`

`Cannot read properties of undefined (reading 'coinbaseRaw')`

And skeleton loader seems not right with the columns
Screen Shot 2022-05-23 at 13 35 07

**Review by wiz — APPROVED** [review]
Tested ACK @ v2.4.0-dev [[aac9dda9](https://github.com/mempool/mempool/commit/aac9dda9)]

Tried all networks, seems fine

**Pull request merged**